tuf/data/roles.go

```diff
@@ -68,6 +68,16 @@ func ValidRole(name string) bool {
 	return false
 }
 
+// IsBaseRole checks if a role name is valid base role
+func IsBaseRole(name string) bool {
```

Comment:
Weren't we going to get rid of this in favour of type checking: `_, ok := r.(BaseRole)`